### PR TITLE
F303- added panning map loads queues from other towns and zooming out a certain

### PR DESCRIPTION
**Pull request by JohnnyPersichilli**
… amount loads small num of queues for the state

### CODE REVIEW

File: CranberryQueue/Map/MapController.swift [hunk 1]
```diff
@@ -64,15 +66,63 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+            let curCoords2D = CLLocation(latitude: curLoc.coordinate.latitude, longitude: curLoc.coordinate.longitude)
+            getGeoCode(withLocation: curCoords2D)
         }
         mapControllerDelegate?.toggleDetailModal(withData: marker.userData as! CQLocation)
         return true
     }
+    
+    func mapView(_ mapView: GMSMapView, idleAt position: GMSCameraPosition) {
+        self.currZoom = mapView.camera.zoom
+        print("map zoom is ",String(self.currZoom))
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: nice. delete this print tho

File: CranberryQueue/Map/MapController.swift [hunk 2]
```diff
@@ -64,15 +66,63 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+        
+        let centerLat = mapView.camera.target.latitude as CLLocationDegrees
+        let centerLong = mapView.camera.target.longitude as CLLocationDegrees
+        let circleCenter = CLLocation(latitude: centerLat, longitude: centerLong)
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: can this be simplified by just passing mapView.camera.target?

File: CranberryQueue/Map/MapController.swift [hunk 3]
```diff
@@ -64,15 +66,63 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+        getGeoCode(withLocation: circleCenter)
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: or passing the loc by attribute?

File: CranberryQueue/Map/MapController.swift [hunk 4]
```diff
@@ -181,10 +231,11 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
         let coder = CLGeocoder()
         coder.reverseGeocodeLocation(loc) { (marks, error) in
             guard let res = marks else {
-                print(error!)
+                print("Geo code err:", error!)
                 return
             }
             self.mapControllerDelegate?.updateGeoCode(city: res[0].locality!, region: res[0].administrativeArea!)
+            print("new city", res[0].locality!)
             self.watchLocationQueues(city: res[0].locality!, region: res[0].administrativeArea!)
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: is this memory safe? does it remove the old listener?

File: CranberryQueue/Map/MapViewController.swift [hunk 5]
```diff
@@ -346,18 +347,33 @@ class MapViewController: UIViewController, UITextFieldDelegate, MapControllerDel
...
+                        "city": currCity,
+                        "region": currRegion,
+                        "numMembers": 0,
+                        "currentSong": "",
+                        "name" : name
+                        ])
+                    let name = self.createQueueForm.queueNameTextField.text!
+                    self.presentQueueScreen(queueId: id, name: name, code: nil, isHost: true)
+                }
+            }
```
Comment (rolflocher) [inline CranberryQueue/Map/MapViewController.swift]: can we do this without importing google maps? i think we have a delegate method to get the geocode

File: CranberryQueue/Map/MapController.swift (hunk 4 above)
Comment (JohnnyPersichilli) [reply CranberryQueue/Map/MapController.swift]: i was thinking about this and i think it overwrites the old listerner which doesnt change its life cycle

Comment (rolflocher) [reply CranberryQueue/Map/MapController.swift]: also remove all prints for sure

File: CranberryQueue/Map/MapController.swift (hunk 2 above)
Comment (JohnnyPersichilli) [reply CranberryQueue/Map/MapController.swift]: the target returns a CLLocation2D but the other function needs a CLLocation. I did simplify in next push tho

File: CranberryQueue/Map/MapViewController.swift (hunk 5 above)
Comment (JohnnyPersichilli) [reply CranberryQueue/Map/MapViewController.swift]: yeh changed it to do it without

File: CranberryQueue/Base.lproj/Main.storyboard [hunk 6]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="14868" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="H7q-7M-o7m">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="15504" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="H7q-7M-o7m">
     <device id="retina5_9" orientation="portrait" appearance="light"/>
     <dependencies>
         <deployment identifier="iOS"/>
-        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="14824"/>
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="15508"/>
```
[the remark this answers is not in the document]
Comment (rolflocher) [reply CranberryQueue/Base.lproj/Main.storyboard]: git checkout master /pathToStoryBoardFile

^ will undo an individual files diff in the future

File: CranberryQueue/Map/MapController.swift [hunk 7]
```diff
@@ -64,15 +66,64 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+        )
+        
+        getGeoCode(withLocation: mapCenter)
+    }
 
     func watchLocationQueues(city: String, region: String) {
+        if(self.currZoom > 10){
+            getTopQueusInCity(city: city, region: region)
+        }else{
+            getTopQueusInState(region: region)
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: nice

File: CranberryQueue/Map/MapController.swift [hunk 8]
```diff
@@ -64,15 +66,64 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+                    long: doc.data()["long"] as! Double,
+                    lat: doc.data()["lat"] as! Double,
+                    queueId: doc.documentID,
+                    numMembers: doc.data()["numMembers"] as! Int
+                )
+                    
+                self.queues.append(newLoc)
+            }
+            self.drawMarkers()
+
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: delete deez whitespaces

File: CranberryQueue/Map/MapController.swift [hunk 9]
```diff
@@ -181,21 +232,41 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
         let coder = CLGeocoder()
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: can you write this function with a completion handler? bc it is not very clear that getGeoCode() is continuing the lifecycle

File: CranberryQueue/Map/MapController.swift [hunk 10]
```diff
@@ -181,21 +232,41 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
         }
+        var currCity = ""
+        var currRegion = ""
+        
+        let currPosition = CLLocation(
+            latitude: curCoords?.latitude ?? 0,
+            longitude: curCoords?.longitude ?? 0
+        )
+        let coder = CLGeocoder()
+        coder.reverseGeocodeLocation(currPosition) { (marks, error) in
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: this getCoords function gets called a lot. we probably shouldnt automatically geocode?

File: CranberryQueue/Map/MapController.swift [hunk 11]
```diff
@@ -181,21 +232,41 @@ class MapController: UIViewController, CLLocationManagerDelegate, GMSMapViewDele
...
+            }
+            currCity = res[0].locality!
+            currRegion = res[0].administrativeArea!
+        }
         return [
             "long": curCoords?.longitude ?? 0,
-            "lat": curCoords?.latitude ?? 0
+            "lat": curCoords?.latitude ?? 0,
+            "city": currCity,
+            "region": currRegion
```
Comment (rolflocher) [inline CranberryQueue/Map/MapController.swift]: this function has been changed from its og functionality. you should move the functionality to leverage the existing getGeoCode func or rename the func. Not a fan of getCoords returning a dic of [String:Any]

**Comment by rolflocher** [comment]
It seems like you are making these changes with an unobtrusive mindset, which I should warn you against. When you tack on new functionality to existing structures, the result isnt always as maintainable as it is set out to be. Sometimes you have to take the existing functionality and make it your own. eg. changing getCoords to return coords and geocode info. In that case you should either change the function's name to reflect new functionality or figure out how we were getting the geocode results without it. Leaving it with the obsolete name isn't fair to other devs. This is a big feature but we shouldnt implement it as if we will never revisit mapCon. Especially bc this is changing the lifecycle we want the results to be super legible

**Pull request merged**